examples/kafka-hub/hub/websub_subscribers.bal

```diff
@@ -0,0 +1,164 @@
+        // if the subscription already exists in the `hub` instance, or the given subscription
+        // does not belong to the `hub` instance do not start the consumer
+        if subscriberAvailable || serverId != config:SERVER_ID {
+            continue;
+        }
+        kafka:Consumer consumerEp = check conn:createMessageConsumer(namespaceId, consumerGroup);
+        _ = check consumerEp->assign([{topic: consumerGroupMapping.eventHub, partition: consumerGroupMapping.partition}]);
+        websubhub:HubClient hubClientEp = check new (subscriber, {
+            retryConfig: {
+                interval: config:MESSAGE_DELIVERY_RETRY_INTERVAL,
+                count: config:MESSAGE_DELIVERY_COUNT,
+                backOffFactor: 2.0,
+                maxWaitInterval: 20
+            },
+            timeout: config:MESSAGE_DELIVERY_TIMEOUT
+        });
+        string topicName = util:sanitizeTopicName(subscriber.hubTopic);
+        _ = @strand {thread: "any"} start pollForNewUpdates(hubClientEp, consumerEp, topicName, subscriberId);
+    }
+}
+
+isolated function pollForNewUpdates(websubhub:HubClient clientEp, kafka:Consumer consumerEp, string topicName, string subscriberId) {
+    do {
+        while true {
+            types:UpdateMessageConsumerRecord[] records = check consumerEp->poll(config:POLLING_INTERVAL);
+            if !isValidConsumer(topicName, subscriberId) {
+                fail error(string `Subscriber with Id ${subscriberId} or topic ${topicName} is invalid`);
+            }
+            _ = check notifySubscribers(records, clientEp, consumerEp);
+        }
+    } on fail var e {
+        lock {
+            _ = subscribersCache.removeIfHasKey(subscriberId);
+        }
+        log:printError("Error occurred while sending notification to subscriber", err = e.message());
+
+        kafka:Error? result = consumerEp->close(config:GRACEFUL_CLOSE_PERIOD);
+        if result is kafka:Error {
+            log:printError("Error occurred while gracefully closing kafka-consumer", err = result.message());
+        }
+    }
+}
+
+isolated function isValidConsumer(string topicName, string subscriberId) returns boolean {
+    boolean subscriberAvailable = true;
+    lock {
+        subscriberAvailable = subscribersCache.hasKey(subscriberId);
+    }
+    return isValidTopic(topicName) && subscriberAvailable;
+}
+
+isolated function notifySubscribers(types:UpdateMessageConsumerRecord[] records, websubhub:HubClient clientEp, kafka:Consumer consumerEp) returns error? {
+    foreach types:UpdateMessageConsumerRecord kafkaRecord in records {
+        websubhub:ContentDistributionMessage message = {
+            content: kafkaRecord.value,
+            contentType: mime:APPLICATION_JSON
+        };
+        websubhub:ContentDistributionSuccess|error response = clientEp->notifyContentDistribution(message);
+        if response is error {
+            return response;
+        }
+        _ = check consumerEp->commit();
```

Comment:
we can commit after the forloop

Reply:
Fixed with [d70caa7ea183e524de4ebe4c658c4e8e368c6d49](https://github.com/ballerina-platform/module-ballerina-websubhub/pull/716/commits/d70caa7ea183e524de4ebe4c658c4e8e368c6d49)